drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -690,6 +697,21 @@ static int nrf5_configure(const struct device *dev,
 	case IEEE802154_CONFIG_EVENT_HANDLER:
 		nrf5_data.event_handler = config->event_handler;
 
+#if defined(CONFIG_OPENTHREAD_THREAD_VERSION_1_2)
```

Comment:
I think it would make sense to introduce a new Kconfig at the shim-layer level, enabled by default if OpenThread 1.2 is used.

Reply:
Added `NET_PKT_TXSEC`, but probably it should go somewhere else.

Reply:
I was actually thinking to add it [here](https://github.com/nrfconnect/sdk-zephyr/blob/master/drivers/ieee802154/Kconfig.nrf5). It's more of a driver feature than net_pkt one.

Reply:
Given that it's used in the platform implementation (`radio.c`) I guess it would be better in https://github.com/nrfconnect/sdk-zephyr/blob/master/drivers/ieee802154/Kconfig.

Reply:
For the `radio.c` it's fine to use `CONFIG_OPENTHREAD_THREAD_VERSION_1_2` since it's a part of the OT implementation. I was thinking of introducing a separate config just for the radio driver (which could default to `y` if OT 1.2 is used). In case it's disabled in the radio driver, the configuration would simply fail. I see though the error is ignored right now and since it cannot be propagated it would make sense IMO to add an assert in `radio.c` in case the configuration fails. This would also be a clear indication for the other drivers that they need to implement this feature in order to use OT 1.2.

Reply:
Actually now I think the new Kconfig to be introduced by Radio Driver to support this functionality should be used here.